com.unity.multiplayer.mlapi/Runtime/Core/NetworkObject.cs

```diff
@@ -598,5 +599,169 @@ internal NetworkBehaviour GetNetworkBehaviourAtOrderIndex(ushort index)
+                var currentStreamPosition = buffer.Position;
+
+                // Write variable data size placeholder initially as zero
+                writer.WriteUInt32(0);
+
+                //Write the network variable data
+                WriteNetworkVariableData(writer.GetStream(), targetClientId);
+
+                //As long as something was written
+                if (buffer.Position > currentStreamPosition)
+                {
+                    //Get the new updated stream buffer position
+                    var updatedPosition = buffer.Position;
+
+                    //Move back to the position just before we wrote our size
+                    buffer.Position = currentStreamPosition;
+
+                    // Replace the zero value placeholder with the size of data written into the stream by WriteNetworkVariableData
+                    writer.WriteUInt32((uint)(buffer.Position - currentStreamPosition));
+
+                    //Revert the buffer position back to the end of the network variable data written
+                    buffer.Position = updatedPosition;
+                }
+            }
+        }
+
+        /// <summary>
+        /// Ueed to deserialize a serialized scene object which occurs
+        /// when the client is approved or during a scene transition
+        /// </summary>
+        /// <param name="objectStream">inbound stream</param>
+        /// <param name="reader">reader for the stream</param>
+        /// <param name="networkManager">NetworkManager instance</param>
+        internal static void DeserializeSceneObject(NetworkBuffer objectStream, NetworkReader reader, NetworkManager networkManager)
+        {
+            var isPlayerObject = reader.ReadBool();
+            var networkId = reader.ReadUInt64Packed();
+            var ownerClientId = reader.ReadUInt64Packed();
+            var hasParent = reader.ReadBool();
+            ulong? parentNetworkId = null;
+
+            if (hasParent)
+            {
+                parentNetworkId = reader.ReadUInt32Packed();
+            }
+
+            var isSceneObject = reader.ReadBool();
+
+            var prefabHash = reader.ReadUInt32Packed();
+            Vector3? position = null;
+            Quaternion? rotation = null;
+
+            // Check to see if we have position and rotation values that follows
+            if (reader.ReadBool())
+            {
+                position = new Vector3(reader.ReadSinglePacked(), reader.ReadSinglePacked(), reader.ReadSinglePacked());
+                rotation = Quaternion.Euler(reader.ReadSinglePacked(), reader.ReadSinglePacked(), reader.ReadSinglePacked());
+            }
+
+            var networkObject = networkManager.SpawnManager.CreateLocalNetworkObject(isSceneObject, prefabHash, ownerClientId, parentNetworkId, position, rotation);
+
+            //Determine if we have network variable data to read
```

Comment:
Ok, this section seems new.  That's cool, I just don't know what's going on to review it.

Reply:
This makes sure that if one NetworkObject fails to construct (for whatever reason) that it can "skip past" that specific NetworkObject but continue processing any remaining serialized NetworkObjects as opposed to just throwing an exception and skipping the remaining (if any) NetworkObjects which was the original behavior.

Reply:
Guess what I'm going to ask you to do :)

_let's add the paragaph you just typed for me into the code_